src/backend/postmaster/pgarch.c

```diff
@@ -577,6 +578,14 @@ pgarch_archiveXlog(char *xlog)
 					strlcpy(dp, contentid, endp - dp);
 					dp += strlen(dp);
 					break;
+				case 'd':
```

Comment:
contentId and dbid have the same type - int32. What do you think about processing them in one case? The new dbid variable will not be required. 
```
				case 'c': /* GPDB: %c: contentId of segment */
                case 'd': /* GPDB: %d: dbid of segment */
                {
                    char    buf[12];  /* sign, 10 digits and '\0' */
                    int32   val = (sp[1] == 'c') ? GpIdentity.segindex : GpIdentity.dbid;

                    Assert(val != UNINITIALIZED_GP_IDENTITY_VALUE);
                    sp++;
                    pg_ltoa(val, buf);
                    strlcpy(dp, buf, endp - dp);
                    dp += strlen(dp);
                    break;
                }
```

Reply:
done